CosmicReco/src/SimpleTimeCluster_module.cc

```diff
@@ -1,231 +1,261 @@
+    _shfToken(conf().shfToken()), _hsel(conf().hsel()), _hbkg(conf().hbkg()) {
+  produces<TimeClusterCollection>();
+}
 
-    if(_testflag){
-      auto shfH = event.getValidHandle<StrawHitFlagCollection>(_shfToken);
-      _shfcol = shfH.product();
-      if(_shfcol->size() != _chcol->size())
-        throw cet::exception("RECO")<<"SimpleTimeCluster: inconsistent flag collection length " << endl;
-    }
+//--------------------------------------------------------------------------------------------------------------
+void SimpleTimeCluster::produce(art::Event& event) {
+  _iev = event.id().event();
+
+  auto const& chH = event.getValidHandle<ComboHitCollection>(_chToken);
+  _chcol = chH.product();
+
+  if (_testflag) {
+    auto shfH = event.getValidHandle<StrawHitFlagCollection>(_shfToken);
+    _shfcol = shfH.product();
+    if (_shfcol->size() != _chcol->size())
+      throw cet::exception("RECO")
+          << "SimpleTimeCluster: inconsistent flag collection length " << std::endl;
+  }
 
-    std::unique_ptr<TimeClusterCollection> tccol(new TimeClusterCollection);
-    findClusters(*tccol);
+  std::unique_ptr<TimeClusterCollection> tccol(new TimeClusterCollection);
+  findClusters(*tccol);
 
-    if (_debug > 0) std::cout << "Found " << tccol->size() << " Time Clusters " << std::endl;
+  if (_debug > 0)
+    std::cout << "Found " << tccol->size() << " Time Clusters " << std::endl;
 
-    if (_debug > 1){
-      for(auto const& tc : *tccol) {
-        std::cout << "Time Cluster time = " << tc.t0().t0() << " +- " << tc.t0().t0Err()
-          << " position = " << tc._pos << std::endl;
-        if(_debug > 3){
-          for (auto shi : tc._strawHitIdxs ) {
-            std::cout << "Time Cluster hit at index " << shi << std::endl;
-          }
+  if (_debug > 1) {
+    for (auto const& tc : *tccol) {
+      std::cout << "Time Cluster time = " << tc.t0().t0() << " +- " << tc.t0().t0Err()
+                << " position = " << tc._pos << std::endl;
+      if (_debug > 3) {
+        for (auto shi : tc._strawHitIdxs) {
+          std::cout << "Time Cluster hit at index " << shi << std::endl;
         }
       }
     }
-    event.put(std::move(tccol));
   }
+  event.put(std::move(tccol));
+}
 
-  void SimpleTimeCluster::findClusters(TimeClusterCollection& tccol) {
-    //sort the hits by time
-    ComboHitCollection ordChCol;
-    ordChCol.reserve(_chcol->size());
+void SimpleTimeCluster::findClusters(TimeClusterCollection& tccol) {
+  // sort the hits by time
+  ComboHitCollection ordChCol;
+  ordChCol.reserve(_chcol->size());
 
-    for (size_t i=0; i<_chcol->size(); ++i) {
-      const ComboHit& ch  = _chcol->at(i);
-      if (_testflag && !goodHit((*_shfcol)[i]))
-        continue;
-      ordChCol.push_back(ComboHit(ch));
+  for (size_t i = 0; i < _chcol->size(); ++i) {
+    const ComboHit& ch = _chcol->at(i);
+    if (_testflag && !goodHit(ch.flag()))
```

Comment:
Looks like you should remove _shfcol and related fcl configuration

Reply:
Thanks! fixed